mm2src/lp_ordermatch.rs

```diff
@@ -3678,6 +3700,36 @@ impl OrderbookP2PItem {
         }
     }
 
+    fn as_rpc_best_orders_buy_v2(
+        &self,
+        address: OrderbookAddress,
+        conf_settings: Option<&OrderConfirmationsSettings>,
+        is_mine: bool,
+    ) -> RpcOrderbookEntryV2 {
+        let price_mm = MmNumber::from(self.price.clone());
+        let max_vol_mm = MmNumber::from(self.max_volume.clone());
+        let min_vol_mm = MmNumber::from(self.min_volume.clone());
+
+        let base_max_volume = max_vol_mm.clone().into();
+        let base_min_volume = min_vol_mm.clone().into();
+        let rel_max_volume = (&max_vol_mm * &price_mm).into();
+        let rel_min_volume = (&min_vol_mm * &price_mm).into();
+
+        RpcOrderbookEntryV2 {
+            coin: self.rel.clone(),
+            address,
+            price: price_mm.into(),
+            pubkey: self.pubkey.clone(),
+            uuid: self.uuid,
+            is_mine,
+            base_max_volume,
+            base_min_volume,
+            rel_max_volume,
+            rel_min_volume,
+            conf_settings: conf_settings.cloned(),
+        }
```

Comment:
This can be shorter
```rust
let price_mm = MmNumber::from(self.price.clone());
        let max_vol_mm = MmNumber::from(self.max_volume.clone());
        let min_vol_mm = MmNumber::from(self.min_volume.clone());

        RpcOrderbookEntryV2 {
            coin: self.rel.clone(),
            address,
            price: price_mm.clone().into(),
            pubkey: self.pubkey.clone(),
            uuid: self.uuid,
            is_mine,
            base_max_volume: max_vol_mm.clone().into(),
            base_min_volume: min_vol_mm.clone().into(),
            rel_max_volume: (&max_vol_mm * &price_mm).into(),
            rel_min_volume: (&min_vol_mm * &price_mm).into(),
            conf_settings: conf_settings.cloned(),
        }
```